18 bookdeatails component

olgakahraman opened the pull request:
Refactored Book details by creating a new folder and moving related functions there. Took control over cover image display in BookCard. Displayed content from BookDetailsMain in App.jsx. Removed BookCover component due to issues with css module access. Made components more reusable by removing extra classNames, then added CSS module classNames for better styling control.

ElijahMSmith commented on public/images/img-1.jpeg:
It looks like you have lots of commits here from #1, which makes it hard for a reviewer to know what new changes are in this PR.

Let's get that merged in and squashed first, and then let's rebase this change on top of that commit. That way, only new code in this PR will get merged after you make the other changes. Again, let me know if you need help here.

ElijahMSmith commented on src/components/BookCard/BookCardBody.jsx:
```diff
@@ -0,0 +1,21 @@
+	author,
+}) {
+	return (
+		<div className="card-body">
```
Same comment as the other PR, but I would expect all the HTML and CSS for this component to exist in one component (BookCard.jsx) so that it's easy to read and make changes.

ElijahMSmith commented on src/components/BookCard/BookCardMain.jsx:
```diff
@@ -0,0 +1,24 @@
+	title,
+	publishDate,
+	quote,
+  author
```
This component needs to be formatted.

ElijahMSmith commented on src/components/BookDetails/BookDetailsMain.jsx:
```diff
@@ -0,0 +1,48 @@
+	return (
+		<div className={styles.bookDetailsContainer}>
+			<div className={styles.cover}>
+				<BookCover coverImg={coverImg} />
```
These components are all one-liners and not worth re-using since their CSS is likely going to change in the future anyways. Let's keep all the HTML and CSS in this component.

ElijahMSmith commented on src/components/BookDetails/BookTitle.jsx:
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+export default function BookTitle({ title }) {
+	return <p className="book-title"> {title}</p>;
```
Should there be an extra space before this title?

olgakahraman commented:
Refactored components and cleaned up code

- Removed unnecessary components and CSS modules.
- Simplified the structure and kept only the essential components.

ElijahMSmith commented on src/components/BookDetails/BookDetails.jsx:
```diff
@@ -0,0 +1,32 @@
+					By {authors} ({publishDate})
+				</p>
+				<p className="book-genre">
+					<span>{categories}</span>
```
Nitpick: Do we need a span if all the text is in that span? I think we could probably just style the `.book-genre` element instead.

ElijahMSmith commented on src/util/index.js:
```diff
@@ -21,5 +21,18 @@ import axios from "axios";
     }
   };
 
+  const [message, setMessage] = useState('');
```
None of this code is useful (the message state or the useEffect below) and therefore I think we should remove it. State is only useful inside of a component, which it is not here.
